drivers/eks/eks_driver.go

```diff
@@ -587,7 +587,7 @@ func (d *Driver) Create(ctx context.Context, options *types.DriverOptions, _ *ty
 	keyPairName := state.KeyPairName
 
 	if keyPairName == "" {
-		keyPairName := getEC2KeyPairName(state.DisplayName)
+		keyPairName = getEC2KeyPairName(state.DisplayName)
```

Comment:
This is the source of previous errors. This variable was shadowing the outer context.